### PR TITLE
Fix(#major); Curve finance; Add sidechains + update revenue and APR logic

### PR DESCRIPTION
**Pull request by harsh9200**
**APR Formula**:
- Mainnet:
```
Staked_Output_Token = GAUGE_WORKING_SUPPLY
Daily_Reward_Emissions =  INFLATION_RATE * GAUGE_RELATIVE_WEIGHT * BALANCER_TOKEN_PRICE * 86_400

APR = (Daily_Reward_Emissions / (TVL) * (Staked_Output_Token / Output_Token_Supply)) * 100 * 365
```
- Sidechain ([NetworksInfo](https://gist.github.com/harsh9200/0afbe5d31499bed5abd934bfcc695c41))
```
Staked_Output_Token = GAUGE_TOTAL_SUPPLY
Daily_Reward_Emissions =  INFLATION_RATE_OF_CURRENT_WEEK * BALANCER_TOKEN_PRICE * 86_400 * 0.4

APR = (Daily_Reward_Emissions / (TVL) * (Staked_Output_Token / Output_Token_Supply)) * 100 * 365
```

Added following chains support to Curve:

- [x] [Arbitrum](https://thegraph.com/hosted-service/subgraph/harsh9200/curve-finance-arbitrum)
- [x] [Aurora](https://thegraph.com/hosted-service/subgraph/harsh9200/curve-finance-aurora)
- [x] [Avalanche](https://thegraph.com/hosted-service/subgraph/harsh9200/curve-finance-avalanche)
- [x] [Fantom](https://thegraph.com/hosted-service/subgraph/harsh9200/curve-finance-fantom)
- [x] [Gnosis](https://thegraph.com/hosted-service/subgraph/harsh9200/curve-finance-gnosis)
- [x] [Harmony](https://thegraph.com/hosted-service/subgraph/harsh9200/curve-finance-harmony)
- [x] [Moonbeam](https://thegraph.com/hosted-service/subgraph/harsh9200/curve-finance-moonbeam)
- [x] [Optimism](https://thegraph.com/hosted-service/subgraph/harsh9200/curve-finance-optimism)
- [x] [Polygon](https://thegraph.com/hosted-service/subgraph/harsh9200/curve-finance-polygon)

### CODE REVIEW

File: subgraphs/curve-finance/src/common/constants.ts [hunk 1]
```diff
@@ -95,6 +95,10 @@ export namespace NULL {
 export namespace Protocol {
```
Comment (jaimehgb) [inline subgraphs/curve-finance/src/common/constants.ts:95]: Doesn’t this file need to be deleted entirely?

File: subgraphs/curve-finance/src/modules/Deposit.ts [hunk 2]
```diff
@@ -41,19 +37,19 @@ export function createDepositTransaction(
...
 
-    depositTransaction.to = liquidityPool.id;
+    depositTransaction.to = pool.id;
     depositTransaction.from = provider.toHexString();
 
     depositTransaction.hash = transaction.hash.toHexString();
     depositTransaction.logIndex = transaction.index.toI32();
 
-    depositTransaction.inputTokens = inputTokens;
+    depositTransaction.inputTokens = pool._inputTokensOrdered;
```
Comment (jaimehgb) [inline subgraphs/curve-finance/src/modules/Deposit.ts:49]: Will this work? `Deposit.inputTokens` is ` [Token]`, so when querying the order will change

**Review by jaimehgb — COMMENTED** [review]
Overall looks good to me. Double check the comment on tokens order on deposits/withdrawals and swaps. Other than that I think its fine!

**Comment by jaimehgb** [comment]
Polygon failed to sync btw :(

**Comment by jaimehgb** [comment]
Look at swap `swap-0x0000656acd8d8256f9847f83a2de71d0abf11be7a1e5598e9efa47bd357555aa-1` on fantom. Tokens don’t match the actual ones that got swapped :( 

It definitely has to do with the order of the tokens

File: subgraphs/curve-finance/src/common/constants.ts (hunk 1 above)
Comment (harsh9200) [reply subgraphs/curve-finance/src/common/constants.ts:95]: yeah I will delete this file.

**Review by jaimehgb — APPROVED** [review]

File: subgraphs/curve-finance/src/modules/Deposit.ts (hunk 2 above)
Comment (harsh9200) [reply subgraphs/curve-finance/src/modules/Deposit.ts:49]: yeah, you are right. This won't work.

File: subgraphs/curve-finance/src/common/utils.ts [hunk 3]
```diff
@@ -78,6 +78,54 @@ export function sortByInputTokenOrder(
...
+    rewardTokenEmissionsAmount!,
+    rewardTokenEmissionsUSD!
+  );
+
+  pool.rewardTokens = rewardTokens;
+  pool.rewardTokenEmissionsAmount = rewardTokenEmissionsAmount;
+  pool.rewardTokenEmissionsUSD = rewardTokenEmissionsUSD;
+}
+
+export function multiArraySort(
```
Comment (jaimehgb) [inline subgraphs/curve-finance/src/common/utils.ts:100]: 🔥

**Pull request merged**